pkg/api/user_token.go

```diff
@@ -52,6 +54,81 @@ func (hs *HTTPServer) RevokeUserAuthToken(c *contextmodel.ReqContext) response.R
 	return hs.revokeUserAuthTokenInternal(c, c.UserID, cmd)
 }
 
+func (hs *HTTPServer) RotateUserAuthTokenRedirect(c *contextmodel.ReqContext) response.Response {
+	token := c.GetCookie(hs.Cfg.LoginCookieName)
+	if token == "" {
+		hs.log.FromContext(c.Req.Context()).Debug("Rotate called without session cookie")
+		return response.Redirect(hs.Cfg.AppSubURL + "/login")
+	}
+
+	ip, _ := network.GetIPFromAddress(c.RemoteAddr())
+	res, err := hs.AuthTokenService.RotateToken(c.Req.Context(), auth.RotateCommand{
+		UnHashedToken: token,
+		IP:            ip,
+		UserAgent:     c.Req.UserAgent(),
+	})
+
+	if err != nil {
+		hs.log.FromContext(c.Req.Context()).Debug("Failed to rotate token", "error", err)
+		if errors.Is(err, auth.ErrInvalidSessionToken) {
+			authn.DeleteSessionCookie(c.Resp, hs.Cfg)
+		}
+		return response.Redirect(hs.Cfg.AppSubURL + "/login")
+	}
+
+	if res.UnhashedToken != token {
+		authn.WriteSessionCookie(c.Resp, hs.Cfg, res)
+	}
+
+	return response.Redirect(hs.GetRedirectURL(c))
+}
+
+// swagger:route POST /user/auth-tokens/rotate
+//
+// # Rotate the auth token of the caller
+//
+// Rotate the token of caller, if successful send and new session cookie.
+//
+// Responses:
+// 200: okResponse
+// 401: unauthorisedError
+// 404: notFoundError
+// 500: internalServerError
+func (hs *HTTPServer) RotateUserAuthToken(c *contextmodel.ReqContext) response.Response {
+	token := c.GetCookie(hs.Cfg.LoginCookieName)
```

Comment:
Not sure if there is value factoring what is in common between `RotateeUseAuthTokenRedirect` and `RotateAuthToken`